File: app/lib/model/prediction.dart
import 'package:uuid/uuid.dart';

class Prediction {
  late String _id;
  late DateTime _receivedAt;
  final String _filePath;
  late double _value;

  Prediction(this._filePath, double value, {String? id, DateTime? receivedAt}) {
    _id = id ?? const Uuid().v4();

    _receivedAt = receivedAt ?? DateTime.now();
  }

  String get id => _id;
  DateTime get receivedAt => _receivedAt;
  String get filePath => _filePath;
  double get value => _value;

  set value(double value) => _value;
}

File: web_app/api/model/prediction.py
import time
import datetime
import uuid
import numpy as np
class Prediction:
    uid:str
    filePath:str
    value:float
    receivedAt:time

    def __init__(self, filePath:str, value:float, receivedAt:time=None, uid:str=None):
        if uid == None:
            self.uid = str(uuid.uuid4())
        else:
            self.uid = uid
        if receivedAt == None:
            self.receivedAt = time.time()
        else:
            self.time = time
        self.filePath = filePath
        self.value = np.format_float_positional(value, trim='-')
        self.receivedAt = receivedAt

    def toDict(self):
        return {
            "uid":self.uid,
            "filePath":self.filePath,
            "value":str(self.value),
            "receivedAt": str(datetime.datetime.fromtimestamp(self.receivedAt))
        }